helper/salt/salt.go

```diff
@@ -139,6 +139,12 @@ func (s *Salt) DidGenerate() bool {
 	return s.generated
 }
 
+// SaltIDHashFunc uses the supplied hash function instead of the configured
+// hash func in the salt.
+func (s *Salt) SaltIDHashFunc(id, hashType, delim string, hashFunc HashFunc) string {
+	return hashType + delim + SaltID(s.salt, id, hashFunc)
```

Comment:
I think rather than lock us into formatting here I'd rather clients do their own formatting. Can we just make this take in id and hashFunc?

Reply:
Done.